### PR TITLE
Add 63 geographical domains for .vn ccTLD

### PR DESCRIPTION
**Pull request by trinv**
Add 63 geographical domains for .vn ccTLD

Public Suffix List (PSL) Pull Request (PR) Template
====

Each PSL PR needs to have a description, rationale, indication of DNS validation and syntax checking, as well as a number of acknowledgements from the submitter.  This template must be included with each PR, and the submitting party MUST provide responses to all of the elements in order to be considered.

### Checklist of required steps

* [x] Description of Organization
* [x] Robust Reason for PSL Inclusion
* [x] DNS verification via dig
* [x] Run Syntax Checker (make test)

* [x] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration, and we shall keep the \_PSL txt record in place in the respective zone(s) in the affected section

__Submitter affirms the following:__ 

  * [x] We are listing *any* third-party limits that we seek to work around in our rationale such as those between IOS 14.5+ and Facebook (see [Issue #1245](https://github.com/publicsuffix/list/issues/1245) as a well-documented example)

  * [x] This request was _not_ submitted with the objective of working around other third-party limits

  * [x] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms
  * [x] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting and sorting

---

For Private section requests that are submitting entries for domains that match their organization website's primary domain, please understand that this can have impacts that may not match the desired outcome and take a long time to rollback, if at all.

To ensure that requested changes are entirely intentional, make sure that you read the affectation and propagation expectations, that you understand them, and confirm this understanding. 

PR Rollbacks have lower priority, and the volunteers are unable to control when or if browsers or other parties using the PSL will refresh or update.

(Link: [about propagation/expectations](https://github.com/publicsuffix/list/wiki/Guidelines#appropriate-expectations-on-derivative-propagation-use-or-inclusion))

 * [x] *Yes, I understand*.  I could break my organization's website cookies etc. and the rollback timing, etc is acceptable.  *Proceed*.
---

Description of Organization
====
Vietnam Internet Network Information Center (VNNIC) is an affiliation to the Ministry of Information and Communications, founded on 28th April 2000, and carries out the functions of managing, allocating, supervising and promoting the use of Internet domain name, address (IP), Autonomous System Number (ASN) in Vietnam (collectively called as “Internet resources”); providing Internet-related guidance, statistics on Internet; joining international activities on Internet. VNNIC owns and controls over .vn ccTLD.
I'm a DNS System Engineer of VNNIC (.vn ccTLD)

Organization Website:  https://www.vnnic.vn/

Reason for PSL Inclusion
====
Adding 63 geographical domains for .vn ccTLD:
Second-level Vietnamese domain names shared by administrative boundaries are Vietnamese domain names named after provinces and centrally run cities and used together without being separately granted to any agency or organization. any individual (hanoi.vn, haiphong.vn …);

Number of users this request is being made to serve: Unlimited

DNS Verification via dig
=======
```
dig _psl.angiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.bacgiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.backan.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.baclieu.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.bacninh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.baria-vungtau.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.bentre.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.binhdinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.binhduong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.binhphuoc.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.binhthuan.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.camau.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.cantho.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.caobang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.daklak.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.daknong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.danang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.dienbien.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.dongnai.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.dongthap.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.gialai.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hagiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.haiduong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.haiphong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hanam.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hanoi.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hatinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.haugiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hoabinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.hungyen.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.khanhhoa.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.kiengiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.kontum.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.laichau.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.lamdong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.langson.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.laocai.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.longan.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.namdinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.nghean.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.ninhbinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.ninhthuan.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.phutho.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.phuyen.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.quangbinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.quangnam.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.quangngai.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.quangninh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.quangtri.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.soctrang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.sonla.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.tayninh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.thaibinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.thainguyen.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.thanhhoa.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.thanhphohochiminh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.thuathienhue.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.tiengiang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.travinh.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.tuyenquang.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.vinhlong.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.vinhphuc.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
dig _psl.yenbai.vn TXT +short
"https://github.com/publicsuffix/list/pull/1776"
```

Results of Syntax Checker (`make test`)
=========
```
Making check in tests
  CC       test-is-public.o
  CC       test-is-public-all.o
  CC       test-is-cookie-domain-acceptable.o
  CC       test-is-public-builtin.o
  CC       test-registrable-domain.o
  CCLD     test-is-cookie-domain-acceptable
  CCLD     test-is-public-builtin
  CCLD     test-is-public
  CCLD     test-is-public-all
  CCLD     test-registrable-domain
PASS: test-is-public-builtin
PASS: test-is-public
PASS: test-is-cookie-domain-acceptable
PASS: test-registrable-domain
PASS: test-is-public-all
============================================================================
Testsuite summary for libpsl 0.21.2
============================================================================
# TOTAL: 5
# PASS:  5
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Making check in msvc
```

### CODE REVIEW

File: public_suffix_list.dat [hunk 1]
```diff
@@ -6290,7 +6290,70 @@ name.vn
 net.vn
 org.vn
 pro.vn
-
```
Comment (dnhn) [inline public_suffix_list.dat:6293]: Please add a blank line between lists.

File: public_suffix_list.dat [hunk 2]
```diff
@@ -6290,7 +6290,70 @@ name.vn
...
+thainguyen.vn
+thanhhoa.vn
+thanhphohochiminh.vn
+thuathienhue.vn
+tiengiang.vn
+travinh.vn
+tuyenquang.vn
+vinhlong.vn
+vinhphuc.vn
+yenbai.vn
```
Comment (dnhn) [inline public_suffix_list.dat:6356]: Please do the same here.

**Comment by trinv** [comment]
Hi @dnhn 
Yes, i did. Thanks

**Comment by trinv** [comment]
updated: DNS Verification via dig

**Comment by dnsguru** [comment]
Is there a URL on the official nic site that shows the policy on these that can be included?

**Comment by dnsguru** [comment]
@trinv i notice the addition of §1(q) but it does not list all the geographics, only two of them.  Can you list them all please and then we can validate DNS _psl txt next

**Comment by trinv** [comment]
@dnsguru , ofcourse yes.
I uploaded here: https://vnnic.vn/sites/default/files/tailieu/vn.cctld.domains.txt 
Thanks

**Comment by dnsguru** [comment]
Can this be a permanently available link that you could add a commented
header line for above your pull request?

The objective is to have the public be able to see the connection for the
new entries to a validated, authoritative reference at vnnic.

Thank you for the extra steps, and the attention to the .vn entries is also
appreciated.

**Comment by trinv** [comment]
Ok thanks.
I updated into the PSL.
Please check it.

**Comment by trinv** [comment]
Hi @dnsguru , 
When you have a chance, please please look into this.
Thank you so much !
Tri Nguyen

**Comment by dnsguru** [comment]
Just validating the _psl txt for the NS verification, but thank you for the connection document from VNNIC that explains the listings.  By being able to follow the path from the IANA.ORG website to the .VN NIC page allows for public, third party validation. 

 Having the document that shows the intentional space from the same root domain hosted at the NIC of the .VN subspace, signals the authenticity and authority of the change, complimenting the _psl TXT records in the DNS.
 
 Each are nearly impossible to both be forged by parties other than the delegated manager of .VN, so both being proven makes this a very high integrety pull request.

**Pull request merged**